[tox]
envlist = lint,typing,py3

[testenv]
extras = all
commands = pytest -s -v {posargs} heudiconv

[testenv:lint]
skip_install = True
deps =
    codespell
    flake8
    flake8-bugbear
    flake8-builtins
    flake8-unused-arguments
commands =
    flake8 heudiconv
    codespell

[testenv:typing]
deps =
    mypy
extras = all
commands =
    mypy heudiconv

[pytest]
#  monitor.py requires optional linotify, but would blow tests discovery, does not contain tests within
addopts = --doctest-modules --tb=short --ignore heudiconv/cli/monitor.py
filterwarnings =
    error
    # <https://github.com/enthought/traits/issues/1739>
    ignore:module 'sre_.*' is deprecated:DeprecationWarning:traits
    # pytest generates a number of inscrutable warnings about open files never
    # being closed.  I (jwodder) expect these are due to DataLad not shutting
    # down batch processes prior to garbage collection.
    ignore::pytest.PytestUnraisableExceptionWarning
    # I don't understand why this warning occurs, as we're using six 1.16,
    # which has the named method.
    ignore:_SixMetaPathImporter.find_spec\(\) not found:ImportWarning
    # <https://github.com/sensein/etelemetry-client/pull/44>
    ignore:.*pkg_resources:DeprecationWarning
    # <https://github.com/nipy/nipype/issues/3563>
    ignore:.*Use setlocale.* instead:DeprecationWarning:nipype
    # <https://github.com/nipy/nipype/pull/3670>
    ignore:.*datetime.datetime.utcnow\(\) is deprecated.*:DeprecationWarning:nipype

[coverage:run]
include = heudiconv/*
          setup.py

[flake8]
doctests = True
exclude = .*/,build/,dist/,test/data,venv/
hang-closing = False
unused-arguments-ignore-stub-functions = True
select = A,B,B902,C,E,E242,F,U100,W
ignore = A003,A005,B005,E203,E262,E266,E501,W503

[isort]
atomic = True
force_sort_within_sections = True
honor_noqa = True
lines_between_sections = 1
profile = black
reverse_relative = True
sort_relative_in_force_sorted_sections = True
known_first_party = heudiconv
